### PR TITLE
ci: changed runner labels 

### PR DESCRIPTION
**Pull request by shaeed-noordeen**
There were some workflows that still had the wrong labelled runners or were using githubs runners, we need to change them to org-wire-runners so we have a more coherent understanding of our infrastructure. This pr will be used to understand which workflows require which size or if they need kind cluster. Please check the the runners in the change made and use [this](https://github.com/mesosphere/github-action-runners-deployment#how-to-add-an-org-wide-runner-to-your-repos-workflow) as a criteria. Once the sizes have been confirmed the pr we can merge.

Which issue(s) does this PR fix?:

https://jira.d2iq.com/browse/D2IQ-97191

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/azure-e2e.yaml [hunk 1]
```diff
@@ -19,7 +19,9 @@ jobs:
         - "rocky 9.1"
         buildConfig:
         - "basic"
-    runs-on: self-hosted
+    runs-on: 
+    - self-hosted
+    - small
```
Comment (dkoshkin) [inline .github/workflows/azure-e2e.yaml:24]: What size is selected if not set?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -15,7 +15,9 @@ permissions:
 
 jobs:
   super-linter:
-    runs-on: ubuntu-latest
+    runs-on: 
+    - self-hosted§
```
Comment (dkoshkin) [inline .github/workflows/lint.yml:19]:
```suggestion
    - self-hosted
```

File: .github/workflows/vsphere-e2e.yaml [hunk 3]
```diff
@@ -30,7 +30,9 @@ jobs:
           buildConfig: offline-fips
         - os: "rocky 9.1"
           buildConfig: offline-fips
-    runs-on: self-hosted
+    runs-on: 
+- self-hosted
+- small
```
Comment (dkoshkin) [inline .github/workflows/vsphere-e2e.yaml:35]:
```suggestion
    runs-on: 
    - self-hosted
    - small
```

**Comment by supershal** [comment]
@shaeed-noordeen The jobs are not getting any self-hosted runners. can you please take a look.

**Pull request merged**